src/Primitives/src/StringValues.cs

```diff
@@ -153,13 +213,14 @@ void ICollection<string>.CopyTo(string[] array, int arrayIndex)
 
         private void CopyTo(string[] array, int arrayIndex)
         {
-            if (_values != null)
+            var value = _values;
+            if (value is string[] values)
             {
-                Array.Copy(_values, 0, array, arrayIndex, _values.Length);
+                Array.Copy(values, 0, array, arrayIndex, values.Length);
                 return;
             }
 
-            if (_value != null)
+            if (value != null)
```

Comment:
Split this method? So a short path with copying for `value is string[]` is kept (may inline), and a larger path with `if (value != null)`, which won't inline (ifs, and throwing).

If the larger path is also considered "hot", maybe use _ThrowHelpers_?

Reply:
Done